firmware/controllers/actuators/idle_thread.cpp

```diff
@@ -397,44 +383,26 @@ static percent_t automaticIdleController(float tpsPos, float rpm, int targetRpm,
 	// todo: add 'pidAfterResetDampingPeriodMs' setting
 	errorAmpCoef = interpolateClamped(0.0f, 0.0f, MS2US(/*CONFIG(pidAfterResetDampingPeriodMs)*/1000), errorAmpCoef, timeSincePidResetUs);
 	// If errorAmpCoef > 1.0, then PID thinks that RPM is lower than it is, and controls IAC more aggressively
-	getIdlePid(PASS_ENGINE_PARAMETER_SIGNATURE)->setErrorAmplification(errorAmpCoef);
+	idlePid->setErrorAmplification(errorAmpCoef);
 
-	percent_t newValue = getIdlePid(PASS_ENGINE_PARAMETER_SIGNATURE)->getOutput(targetRpm, rpm, SLOW_CALLBACK_PERIOD_MS / 1000.0f);
+	percent_t newValue = idlePid->getOutput(targetRpm, rpm, SLOW_CALLBACK_PERIOD_MS / 1000.0f);
 	engine->engineState.idle.idleState = PID_VALUE;
 
-	// the state of PID has been changed, so we might reset it now, but only when needed (see idlePidDeactivationTpsThreshold)
-	mightResetPid = true;
-
 	// Apply PID Multiplier if used
 	if (CONFIG(useIacPidMultTable)) {
 		float engineLoad = getFuelingLoad(PASS_ENGINE_PARAMETER_SIGNATURE);
 		float multCoef = iacPidMultMap.getValue(rpm / RPM_1_BYTE_PACKING_MULT, engineLoad);
 		// PID can be completely disabled of multCoef==0, or it just works as usual if multCoef==1
-		newValue = interpolateClamped(0.0f, engine->engineState.idle.baseIdlePosition, 1.0f, newValue, multCoef);
+		newValue = interpolateClamped(0, 0, 1, newValue, multCoef);
 	}
 	
 	// Apply PID Deactivation Threshold as a smooth taper for TPS transients.
 	// if tps==0 then PID just works as usual, or we completely disable it if tps>=threshold
-	newValue = interpolateClamped(0.0f, newValue, CONFIG(idlePidDeactivationTpsThreshold), engine->engineState.idle.baseIdlePosition, tpsPos);
-
-	// Interpolate to the manual position when RPM is close to the upper RPM limit (if idlePidRpmUpperLimit is set).
-	// If RPM increases and the throttle is closed, then we're in coasting mode, and we should smoothly disable auto-pid.
-	// If we just leave IAC at baseIdlePosition (as in case of TPS deactivation threshold), RPM would get stuck. 
-	// That's why there's 'useIacTableForCoasting' setting which involves a separate IAC position table for coasting (iacCoasting).
-	// Currently it's user-defined. But eventually we'll use a real calculated and stored IAC position instead.
-	int idlePidLowerRpm = targetRpm + CONFIG(idlePidRpmDeadZone);
-	if (CONFIG(idlePidRpmUpperLimit) > 0) {
```

Comment:
@andreika-git `idlePidRpmUpperLimit` is at line 215:

```
	// If rpm too high (but throttle not pressed), we're coasting
	int maximumIdleRpm = targetRpm + CONFIG(idlePidRpmUpperLimit);
	if (rpm > maximumIdleRpm) {
		return Phase::Coasting;
	}
```

The problem with this code is that it didn't actually work without the IAC coasting table (which is an open loop correction, what is it doing in the CL logic?).  This is probably related to whatever @abelom was seeing in these issues #2768 #2772

Reply:
And BTW, trying to *taper a closed loop correction* is asking for integrator windup.  Consider this: the engine speed is high, but not enough to leave the idle zone.  Suppose you're right on the edge, so that interpolate ends up knocking out 90% of the correction being made.  You've now given the integrator license to wind up 10x more than it would have without the taper!